modules/video/include/opencv2/video/tracking.hpp

```diff
@@ -265,6 +265,19 @@ enum
     MOTION_HOMOGRAPHY  = 3
 };
 
+/** @brief Computes the Enhanced Correlation Coefficient value between two images @cite EP08 .
+
+@param templateImage single-channel template image; CV_8U or CV_32F array.
+@param inputImage single-channel input image which should be warped with the final warpMatrix in
+order to provide an image similar to templateImage, same type as templateImage.
+@param inputMask An optional mask to indicate valid values of inputImage.
+
+@sa
+findTransformECC
+ */
+
+CV_EXPORTS_W double computeECC(const Mat& templateImage, const Mat& inputImage, const Mat& inputMask);
```

Comment:
1. Use `InputArray` in public API instead of direct Mat.

2.
    > inputMask
    > An optional mask

    inputMask = noArray()

3.
    > which should be warped with the final warpMatrix in
order to provide

    What?
    Re-check parameters documentation and their limitations (CV_8U/32F is not correct).

Reply:
Fixed #1 and #2. Not sure I understand #3. I used the same description for the parameters as the existing findTransformECC function. As long as both images are of the same type, and are single channel, we should be good, right?

Reply:
Parameter description uses undefined "warpMatrix" (not used in this function at all).

Reply:
Ah: thank you again. Here's the latest version:

@param inputImage single-channel input image to be warped to provide an image similar to
 templateImage, same type as templateImage.